core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/commandqueue/handlers/CancelTransferCommandHandler.java

```diff
@@ -0,0 +1,40 @@
+package org.eclipse.dataspaceconnector.transfer.core.commandqueue.handlers;
+
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.system.Inject;
+import org.eclipse.dataspaceconnector.spi.transfer.command.CommandHandler;
+import org.eclipse.dataspaceconnector.spi.transfer.store.TransferProcessStore;
+import org.eclipse.dataspaceconnector.transfer.core.commandqueue.commands.CancelTransferCommand;
+
+import static java.lang.String.format;
+
+public class CancelTransferCommandHandler implements CommandHandler<CancelTransferCommand> {
+    @Inject
+    private TransferProcessStore store;
```

Comment:
The handlers will have to use the store, because that's where the modification will take place albeit in a synchronized and deterministic way. 
However, your comment touches on another important question, which is "how do commands get into the system?". There are basically two options: 
1. Use `CommandQueue.enqueue()`
2. Use `TransferProcessManager.enqueueCommand()`

Both of those options suffer the same drawback, i.e. exposing more API surface than necessary. I chose to go with option 2 because then we could make the queue invisible (= not registered in the context) altogether.

Reply:
A clear and concise public API for developers using EDC will definitely help preventing errors, so I would go with the second option too. 

In this PR we are introducing a mechanism to perform operations on TransferProcesses through commands, but it is still possible for an EDC developer to modify TPs using the `TransferProcessStore`. My point was that we should try to reduce the risk of this erroneous usage, if not possible by reducing visibility then by clearly documenting that this is a class internal to the framework.